### PR TITLE
Add ISO4217 Currency Code to Currency Names

### LINKED ISSUE #1032: List of currencies should display ISO 4217 code
This list of currencies is seen on general settings, moderation settings and on onboarding.
image

image

"Euro" should be "Euro (EUR)"; 
"United States Dollar" should be "United States Dollar (USD)"; 

Alternatively it could be displayed this way:
EUR - Euro
USD - United States Dollar

This is useful because enables user to search for the code.

### LINKED ISSUE #1354: Setup selecting base currency does not have GBP ... and should also show currency codes for all
Brief Description: 

Operating System (OS and version): Linux Mint 18.3 (Linux 4.13.0)
OpenBazaar version (shown on `About OpenBazaar` page in menu): 2.2.0 - but can not get the About box at this piont of the install
Hardware: HP Omen 17

Reproducible (Always / Almost Always / Sometimes / Rarely / Couldn't Reproduce): 
Always

Steps to reproduce:
1. Install
2. On the 'set your information' dialogue, select location as United Kingdom Great Britain
3. Try to find base currency GBP under any name or description in drop down

Observed Behavior:
GBP not present

Expected Behavior:
GBP present (and it would help if all the other entries included ISO currency codes)

Additional info (links, images, etc go here):

temp5

### PR DESCRIPTION
**Pull request by jashot7**
closes #1357 
closes #1032 
closes #1354

### CODE REVIEW

File: js/data/currencies.js [hunk 1]
```diff
@@ -683,10 +683,17 @@ function getTranslatedCurrencies(lang = app.localSettings.standardizedTranslated
...
+    const name = app.polyglot.t(`currencies.${currency.code}`);
+    return {
+      ...currency,
+      name,
+      nameWithCode: app.polyglot.t('currencyWithCode', {
+        name,
+        code: currency.code,
+      }),
+    };
+  });
```
Comment (jjeffryes) [inline js/data/currencies.js]: Wouldn't it be simpler to leave currencies.js unchanged, and just call the currencyWithCode in the template, like this:
`<%= ob.polyT('currencyWithCode', { name: currency.name, code: currency.code }) %>`?

File: js/templates/modals/onboarding/onboarding.html [hunk 2]
```diff
@@ -93,7 +93,7 @@
           <% if (ob.settingsErrors.currency) print(ob.formErrorTmpl({ errors: ob.settingsErrors.currency })) %>
           <select id="onboardingCurrency" name="localCurrency" class="clrSh2" data-model="settings">
             <% ob.currencyList.forEach((currency) => { %>
-            <option value="<%= currency.code %>" <% if (currency.code == ob.settings.localCurrency) print('selected') %>><%= currency.name %></option>
+            <option value="<%= currency.code %>" <% if (currency.code == ob.settings.localCurrency) print('selected') %>><%= currency.nameWithCode %></option>
```
Comment (jashot7) [inline js/templates/modals/onboarding/onboarding.html]:
```
currency.nameWithCode
```

is nicer to use across multiple templates and easier to maintain if we need changes than:

```
<%= ob.polyT('currencyWithCode', { name: currency.name, code: currency.code }) %>
```

I definitely prefer the 'logic' being in one place, but I see your point for simplicity.

Comment (jashot7) [reply js/templates/modals/onboarding/onboarding.html]: @jjeffryes ^

File: js/data/currencies.js (hunk 1 above)
Comment (jjeffryes) [reply js/data/currencies.js]: It's just a question/suggestion, not a requirement.

**Pull request merged**